osu.Game/Screens/Play/SoloSpectatorScreen.cs

```diff
@@ -218,7 +224,47 @@ void start()
                 Beatmap.Value = spectatorGameplayState.Beatmap;
                 Ruleset.Value = spectatorGameplayState.Ruleset.RulesetInfo;
 
-                this.Push(new SpectatorPlayerLoader(spectatorGameplayState.Score, () => new SoloSpectatorPlayer(spectatorGameplayState.Score)));
+                OsuScreenStack newStack;
+
+                LoadComponentAsync(new Container
+                {
+                    CornerRadius = 10,
+                    Masking = true,
+                    Alpha = 0,
+                    AlwaysPresent = true,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Both,
+                    Scale = new Vector2(0.95f),
+                    Children = new Drawable[]
+                    {
+                        newStack = new OsuScreenStack()
+                    }
+                }, container =>
+                {
+                    screenStackContainer?.FadeOut(200);
+                    screenStackContainer?.Expire();
+
+                    AddInternal(screenStackContainer = container);
+                    playerScreenStack = newStack;
```

Comment:
I can't really tell if this is safe or not. Especially given this line:

https://github.com/ppy/osu/blob/bce32103ee8a7fe08f66f848052bc91e45ee17db/osu.Game/Screens/Play/SoloSpectatorScreen.cs#L212

Like, what is the guarantee that the following doesn't happen?

- `LoadComponentAsync()` called the first time => `newStack` is assigned the first time
- `LoadComponentAsync()` called the second time => `newStack` is assigned the second time
- second async load completes
- first async load completes => screen stack container from second load is expired
- at which point `playerScreenStack = newStack` happens => playerScreenStack is a child of an expired container